src/lib/core/utils/default-axis-range.ts

```diff
@@ -4,51 +4,54 @@ import { NumberOrDateRange } from '@veupathdb/components/lib/types/general';
 import { ComputedVariableMetadata } from '../api/DataClient/types';
 import { min, max } from 'lodash';
 
-export function numberDateDefaultDependentAxisRange(
-  variable: Variable | undefined,
-  plotName: string,
-  yMinMaxRange:
-    | { min: number | string | undefined; max: number | string | undefined }
-    | undefined,
-  // pass computedVariableMetadata
-  computedVariableMetadata?: ComputedVariableMetadata
+export function numberDateDefaultAxisRange(
+  variable: Variable | ComputedVariableMetadata,
+  /** the min/minPos/max values observed in the data response */
+  observedMin: number | string | undefined,
+  observedMinPos: number | string | undefined,
+  observedMax: number | string | undefined,
+  /** are we using a log scale */
+  logScale?: boolean
 ): NumberOrDateRange | undefined {
-  // make universal range variable
-  if (
-    variable != null &&
-    (plotName === 'scatterplot' || plotName === 'lineplot')
-  ) {
-    // this should check integer as well
+  if (Variable.is(variable)) {
     if (variable.type === 'number' || variable.type === 'integer') {
       const defaults = variable.distributionDefaults;
+      if (logScale && observedMinPos == null) return undefined; // return nothing - there will be no plottable data anyway
```

Comment:
i don't follow - i would have expected `if (logScale && observedMax <= 0)`. Why if the minPos is null then we'd have no plottable data?

Reply:
oh wait i get it. Assume only one point is positive. That point would be the observedMinPos. 
So if there are no minimum positive values there must be no positive values. 👍